packages/remix-dev/compiler/plugins/nativeNodeModules.ts

```diff
@@ -0,0 +1,42 @@
+import type esbuild from "esbuild";
+import * as path from "path";
+// See https://github.com/evanw/esbuild/issues/1051#issuecomment-806325487
+export function nativeNodeModulesPlugin(): esbuild.Plugin {
+    return {
+        name: 'native-node-modules',
+        setup(build) {
+            // If a ".node" file is imported within a module in the "file" namespace, resolve
+            // it to an absolute path and put it into the "node-file" virtual namespace.
+            build.onResolve({ filter: /\.node$/, namespace: 'file' }, args => ({
+                path: require.resolve(args.path, { paths: [args.resolveDir] }),
```

Comment:
Add `sideEffects: false` here to make sure this can be optimized out of client bundles.